src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,54 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INDEX_ZERO = 0;
+    private static final int STORAGE_LENGTH = 10;
+    private final StorageImpl<K, V>[] storage;
```

Comment:
You can create inner class for this case. 
Your solution is hard to read and debug, because of multiple references of object to itself. 

```suggestion
    private final Pair<K, V>[] storage;
    
    static class Pair<K, V> {
    K key; 
    V value;
    }
```